lib/Conversion/TorchToLinalg/TorchToLinalg.cpp

```diff
@@ -761,12 +761,148 @@ static Value createLinalgPayloadCalculationForReduceOp(
   if (isa<AtenSumOp, AtenSumDimIntListOp>(op) &&
       elementType.isa<mlir::FloatType>())
     return b.create<AddFOp>(loc, payloadArgs);
-
   op->emitError("unimplemented lowering in "
                 "createLinalgPayloadCalculationForReduceOp");
   return nullptr;
 }
 
+namespace {
+// Aten argmax lowering represents the ArgMax op as an linalg.indexed_generic
+// op, producing two output buffers.
+//
+// The first output buffer contains the index of the found maximum value. It is
+// initialized to 0 and is resulting integer type.
+//
+// The second output buffer contains the maximum value found. It is initialized
+// to the minimum representable value of the input element type. After being
+// populated by indexed_generic, this buffer is disgarded as only the index is
+// requested.
+//
+// The indexed_generic op updates both the maximum value and index if the
+// current value exceeds the running max.
+class ConvertAtenArgmaxOp : public OpConversionPattern<AtenArgmaxOp> {
+public:
+ using OpConversionPattern<AtenArgmaxOp>::OpConversionPattern;
+
+ LogicalResult matchAndRewrite(AtenArgmaxOp argmaxOp, ArrayRef<Value> operands,
+                  ConversionPatternRewriter &rewriter) const override {
+    
+   auto loc = argmaxOp.getLoc();
+   AtenArgmaxOp::Adaptor adaptor(operands); 
+   Value input = adaptor.self();
+   auto inputType = input.getType().cast<ShapedType>();
+   //auto resultType = argmaxOp.result().getType();
+   auto resultType = getTypeConverter()
+     ->convertType(argmaxOp->getResult(0).getType())
+     .cast<RankedTensorType>();
+   //auto resultType = resultTypeGet.cast<ShapedType>();
+   auto inElementType = inputType.getElementType();
+   auto outElementType = resultType.getElementType();
+   int64_t axis;
+   if (!matchPattern(argmaxOp.dim(), m_TorchConstantInt(&axis)))
+      return rewriter.notifyMatchFailure(argmaxOp, "dim must be constant");
+   auto resultMaxType = RankedTensorType::get(resultType.getShape(), inElementType);
+   if (!inputType.hasStaticShape())
```

Comment:
We never write code that is hardcoded to static shapes in npcomp. Please support dynamic shapes. This generally requires inserting `tensor.dim` ops + math to calculate shapes, then feeding dynamic values to `linalg.init_tensor`.